apps/interpret_nn/compare_vs_tflite.cpp

```diff
@@ -77,25 +76,87 @@ Buffer<void> wrap_tf_lite_tensor_with_halide_buffer(const TfLiteTensor *t) {
     return b;
 }
 
+struct DelegateFactory {
+    TfLiteDelegate *(*create_delegate)(char **options_keys,
+                                       char **options_values,
+                                       size_t num_options,
+                                       void (*report_error)(const char *));
+    void (*destroy_delegate)(TfLiteDelegate *delegate);
+};
+
 }  // namespace
 
-void run_both(const std::string &filename, int seed, int threads, bool verbose) {
+void run_all(const std::string &filename, int seed, int threads, bool verbose, DelegateFactory *delegate_factory) {
     std::cout << "Comparing " << filename << "\n";
 
     std::vector<char> buffer = read_entire_file(filename);
 
-    std::vector<Buffer<const void>> tflite_outputs, halide_outputs;
-    std::chrono::duration<double> tflite_time, halide_time;
+    struct RunResult {
+        std::vector<Buffer<const void>> outputs;
+        std::chrono::duration<double> time;
+    };
+
     std::map<std::string, int> seeds;
 
-    // ----- Run in TFLite
+    RunResult halide_result;
+
+    // ----- Run in Halide
     {
+        Model model = parse_tflite_model_from_buffer(buffer.data());
+        if (verbose) {
+            model.dump(std::cout);
+        }
+
+        ModelInterpreter interpreter(std::move(model));
+
+        // Fill in the inputs with random data (but with the same seeds as above).
+        for (Tensor *t : interpreter.inputs()) {
+            if (t->is_constant()) {
+                // Skip constant buffers, just like TFlite above.
+                continue;
+            }
+            int seed_here = seed++;
+            seeds[t->name()] = seed_here;
+            auto input_buf = t->data<void>();
+            dynamic_type_dispatch<FillWithRandom>(input_buf.type(), input_buf, seed_here);
+            if (verbose) {
+                std::cout << "HALIDE input " << t->name() << " inited with seed = " << seed_here << " type " << input_buf.type() << "\n";
+            }
+        }
+
+        halide_set_num_threads(threads);
```

Comment:
I think it would be better to just assert threads == 1, we don't parallelize yet. We won't parallelize within Halide code either.

Reply:
Not gonna assert because we also use `threads` for `TfLiteInterpreterOptionsSetNumThreads()` -- I'll remove this though and add a comment.